django/contrib/admin/templatetags/admin_list.py

```diff
@@ -312,7 +312,11 @@ def __init__(self, form, *items):
 
 def results(cl):
     if cl.formset:
-        for res, form in zip(cl.result_list, cl.formset.forms):
+        for res, form in zip(
+            cl.result_list[:cl.formset.total_form_count()],
+            cl.formset.forms,
```

Comment:
Similar to the above, in the two cases in this file we only need as many results as there are forms.